rosidl_cmake/rosidl_cmake-extras.cmake

```diff
@@ -31,3 +31,16 @@ include("${rosidl_cmake_DIR}/rosidl_generate_interfaces.cmake")
 include("${rosidl_cmake_DIR}/rosidl_target_interfaces.cmake")
 include("${rosidl_cmake_DIR}/rosidl_write_generator_arguments.cmake")
 include("${rosidl_cmake_DIR}/string_camel_case_to_lower_case_underscore.cmake")
+
+# register ament_package() hook for typesupport libraries once
+macro(_rosidl_cmake_export_typesupport_libraries_register_package_hook)
+  if(NOT DEFINED _ROSIDL_CMAKE_EXPORT_TYPESUPPORT_LIBRARIES_PACKAGE_HOOK_REGISTERED)
+    set(_ROSIDL_CMAKE_EXPORT_TYPESUPPORT_LIBRARIES_PACKAGE_HOOK_REGISTERED TRUE)
+
+    find_package(ament_cmake_core QUIET REQUIRED)
+    ament_register_extension("ament_package" "rosidl_cmake"
+      "rosidl_cmake_export_typesupport_libraries_package_hook.cmake")
+  endif()
+endmacro()
```

Comment:
Should this maco be here? Rather than its own file or in the `rosidl_export_typesupport_libraries.cmake` file?

Reply:
This file is the CMake extra file of the `rosidl_cmake` package. Therefore I just added it there. It could be placed into a second `-extras.cmake` file but it would be rather uncommon that a single CMake package has multiple extra files.